### PR TITLE
C375139 Edit any field in linked MARC authority record without 010 field when 001 = $0 (Spitfire) (TaaS)

### PR DESCRIPTION
**Pull request by ylatysh**
[TestRail](https://foliotest.testrail.io/index.php?/cases/view/375139)
[Allure report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/6270/allure/)

image

### CODE REVIEW

**Review by dnovikau — APPROVED** [review]

File: cypress/e2e/marc/marc-authority/edit-linked-authority-record/marc-authority-edit-linked-without-010-field-when-001-field-value-set-to-0.cy.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    InventoryInstance.deleteInstanceViaApi(createdRecordIDs[0]);
+    MarcAuthority.deleteViaAPI(createdRecordIDs[1]);
+  });
+
+  it(
+    'C375139 Edit any field in linked "MARC authority" record without "010" field when "001" = "$0" (spitfire) (TaaS)',
+    { tags: ['extendedPath', 'spitfire'] },
+    () => {
+      MarcAuthorities.searchAndVerify(testData.searchOption, marcFiles[1].authorityHeading);
+
```
Comment (Sherzod-Kenjaev) [inline cypress/e2e/marc/marc-authority/edit-linked-authority-record/marc-authority-edit-linked-without-010-field-when-001-field-value-set-to-0.cy.js:116]: step 2 - Not all verifications were covered
Screenshot 2023-12-22 163347

File: cypress/e2e/marc/marc-authority/edit-linked-authority-record/marc-authority-edit-linked-without-010-field-when-001-field-value-set-to-0.cy.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      QuickMarcEditor.checkFieldAbsense(testData.tag010);
+
+      QuickMarcEditor.updateExistingField(
+        testData.fieldForEditing.tag,
+        testData.fieldForEditing.newValue,
+      );
+      QuickMarcEditor.checkContent(testData.fieldForEditing.newValue, 7);
+
+      QuickMarcEditor.pressSaveAndClose();
+      QuickMarcEditor.checkAfterSaveAndCloseAuthority();
```
Comment (Sherzod-Kenjaev) [inline cypress/e2e/marc/marc-authority/edit-linked-authority-record/marc-authority-edit-linked-without-010-field-when-001-field-value-set-to-0.cy.js:127]: step 5 - cover all checks please
Screenshot 2023-12-22 163817

File: cypress/e2e/marc/marc-authority/edit-linked-authority-record/marc-authority-edit-linked-without-010-field-when-001-field-value-set-to-0.cy.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+      TopMenuNavigation.navigateToApp('Inventory');
+
+      InventoryInstances.searchAndVerify(marcFiles[0].instanceTitle);
+      InventoryInstances.selectInstance();
```
Comment (Sherzod-Kenjaev) [inline cypress/e2e/marc/marc-authority/edit-linked-authority-record/marc-authority-edit-linked-without-010-field-when-001-field-value-set-to-0.cy.js:132]: step 8 - check required results
Screenshot 2023-12-22 163832

File: cypress/support/fragments/inventory/inventoryInstances.js [hunk 4]
```diff
@@ -279,6 +279,13 @@ export default {
     cy.do(Checkbox(tagName).click());
   },
 
+  searchAndVerify(value) {
+    cy.do(filterSection.find(inventorySearchInput).fillIn(value));
+    cy.expect(filterSection.find(inventorySearchInput).has({ value }));
+    cy.expect(filterSection.find(searchButton).has({ disabled: false }));
```
Comment (Sherzod-Kenjaev) [inline cypress/support/fragments/inventory/inventoryInstances.js:285]: Place cy.expect checks into array please, cy.expect([]);

**Review by Sherzod-Kenjaev — CHANGES_REQUESTED** [review]

**Review by Sherzod-Kenjaev — APPROVED** [review]

File: cypress/e2e/marc/marc-authority/edit-linked-authority-record/marc-authority-edit-linked-without-010-field-when-001-field-value-set-to-0.cy.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      marc: 'marcAuthFileForC375139.mrc',
+      fileName: `testMarcFileC375139.${getRandomPostfix()}.mrc`,
+      jobProfileToRun: 'Default - Create SRS MARC Authority',
+      authorityHeading: 'Beethoven, Ludwig van (no 010)',
+      numOfRecords: 1,
+    },
+  ];
+
+  before('Create test data', () => {
+    cy.loginAsAdmin({
```
Comment (TetianaParanich) [inline cypress/e2e/marc/marc-authority/edit-linked-authority-record/marc-authority-edit-linked-without-010-field-when-001-field-value-set-to-0.cy.js:48]: Please add deleting marc auth record before
cy.getAdminToken();
      // make sure there are no duplicate authority records in the system
      MarcAuthorities.getMarcAuthoritiesViaApi({ limit: 100, query: 'keyword="C375139"' }).then(
        (records) => {
          records.forEach((record) => {
            if (record.authRefType === 'Authorized') {
              MarcAuthority.deleteViaAPI(record.id);
            }
          });
        },
      );
And add to the marcAuthFileForC375139.mrc in the field 100 $a id 'C375139'. It helps deleting duplicates of auth records in DB

**Review by TetianaParanich — APPROVED** [review]

**Pull request merged**